### PR TITLE
docs: update  documentation in snowflake_database resource (#2708)

### PR DESCRIPTION
**Pull request by jacobcbeaudin**
The documentation for the `snowflake_database` resource's `from_share` property has been updated to reflect the necessary changes described in the migration guide v0.86.0 to v0.87.0. This update addresses the modification in the configuration format from `<org_name>.<account_name>` to `<account_locator>`, ensuring consistency with identifier types across the platform.

## References
* issue: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2708
* https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/docs/resources/database.md
* https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0860--v0870

### CODE REVIEW

**Comment by sfc-gh-asawicki** [comment]
Hey @jacobcbeaudin. Thanks for the contribution.

The docs are generated, so a change in the MD file will be removed with the next generation. In this case, you have to change `examples/resources/snowflake_database/resource.tf` and then run `make dev-setup pre-push`. Let me know if you are willing to do that, if not, I will add it to one of my today's PRs.

File: docs/resources/database.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ resource "snowflake_database" "from_share" {
 - `data_retention_time_in_days` (Number) Number of days for which Snowflake retains historical data for performing Time Travel actions (SELECT, CLONE, UNDROP) on the object. A value of 0 effectively disables Time Travel for the specified database. Default value for this field is set to -1, which is a fallback to use Snowflake default. For more information, see [Understanding & Using Time Travel](https://docs.snowflake.com/en/user-guide/data-time-travel).
 - `from_database` (String) Specify a database to create a clone from.
 - `from_replica` (String) Specify a fully-qualified path to a database to create a replica from. A fully qualified path follows the format of `"<organization_name>"."<account_name>"."<db_name>"`. An example would be: `"myorg1"."account1"."db1"`
-- `from_share` (Map of String) Specify a provider and a share in this map to create a database from a share.
+- `from_share` (Map of String) Specify a `provider` and a `share` in this map to create a database from a share. As of version 0.87.0, the `provider` field requires the format of `"<account_locator>"`, which is now case-sensitive.
```
Comment (sfc-gh-asawicki) [inline docs/resources/database.md:60]: Hey @jacobcbeaudin, this part is also generated based on https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/88944e7a02b1ff930e1a39bc519780c124bdf4b2/pkg/resources/database.go#L44, so please update the parameter description in resource definition

**Comment by jacobcbeaudin** [comment]
> Hey @jacobcbeaudin. Thanks for the contribution.
> 
> The docs are generated, so a change in the MD file will be removed with the next generation. In this case, you have to change `examples/resources/snowflake_database/resource.tf` and then run `make dev-setup pre-push`. Let me know if you are willing to do that, if not, I will add it to one of my today's PRs.

Done

File: docs/resources/database.md (hunk 1 above)
Comment (jacobcbeaudin) [reply docs/resources/database.md]: Updated

**Comment by sfc-gh-asawicki** [comment]
Merged, thanks for the contribution @jacobcbeaudin!

**Pull request merged**